### PR TITLE
feat: add plugin from treeview

### PR DESCRIPTION
**Pull request by yuqizhou77**
- Add plugin from treeview. User can add a new plugin to a declarative copilot from either an API spec or existing plugin manifest.
- Adjust CLI help to not show DC related questions if feature flag is not on.
- Fix an issue in VSC that the config key does not update all env which lead users will see new features if the env variables was set on machine but the user settings in VSC is off.

### CODE REVIEW

File: packages/fx-core/resource/package.nls.json [hunk 1]
```diff
@@ -442,13 +442,12 @@
...
   "core.SampleSelect.buttons.viewSamples": "View samples",
-  "core.question.pluginAvailability.title": "Select Plugin Availability",
-  "core.pluginAvailability.declarativeCopilot": "Declarative Copilot",
-  "core.pluginAvailability.copilotForM365": "Copilot for Microsoft 365",
-  "core.pluginAvailability.declarativeCopilotAndM365": "Both declarative Copilot and Copilot for Microsoft 365",
-  "core.addPlugin.success": "Plugin \"%s\" added to the project successfully.",
-  "core.addAction.success": "Action \"%s\" added to the project successfully.",
-  "core.addActionAndPlugin.success": "Action \"%s\" and plugin \"%s\" added to the project successfully.",
+  "core.addPlugin.success.vsc": "API plugin \"%s\" added to the project successfully.",
+  "core.addPlugin.success": "API plugin \"%s\" added to the project successfully. You can view plugin manifest in \"%s\".",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:446]: API plugin \"%s\" added to the project successfully. View plugin manifest in \"%s\"

File: packages/fx-core/resource/package.nls.json [hunk 2]
```diff
@@ -442,13 +442,12 @@
...
+  "core.addPlugin.success.viewPluginManifest": "View plugin manifest",
+  "core.scaffold.warning.summary": "We have detected the following issues:\n%s",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:448]: Remove 'the'.

File: packages/fx-core/resource/package.nls.json [hunk 3]
```diff
@@ -442,13 +442,12 @@
...
+  "core.addPlugin.warning.manifestVariables": "Found environment variables \"%s\" in the manifest of the added plugin. Ensure the values are set in .env file or system environment variables.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:449]: Environment variables \"%s\" found in manifest of the added plugin. Ensure the values are set in .env file or system environment variables.

File: packages/fx-core/resource/package.nls.json [hunk 4]
```diff
@@ -442,13 +442,12 @@
...
+  "core.addPlugin.warning.apiSpecVariables": "Found environment variables \"%s\" in the API specification of the added plugin. Ensure the values are set in .env file or system environment variables.",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:450]: Environment variables \"%s\" found in API specification of the added plugin. Ensure the values are set in .env file or system environment variables.

**Review by jayzhang — APPROVED** [review]

**Review by HuihuiWu-Microsoft — APPROVED** [review]

**Review by huimiu — APPROVED** [review]

**Review by tecton — APPROVED** [review]

File: packages/fx-core/resource/package.nls.json [hunk 5]
```diff
@@ -388,18 +388,18 @@
   "core.createProjectQuestion.apiSpec.operation.placeholder.skipExisting": "Methods defined in manifest.json are not listed",
   "core.createProjectQuestion.apiSpec.multipleValidationErrors.message": "Incompatible OpenAPI description document. Check output panel for details.",
   "core.createProjectQuestion.apiSpec.multipleValidationErrors.vscode.message": "Incompatible OpenAPI description document. Check [output panel](command:fx-extension.showOutputChannel) for details.",
-  "core.createProjectQuestion.meArchitecture.title": "Architecture of search based message extension",
-  "core.createProjectQuestion.declarativeCopilot.title": "Create declarative copilot",
+  "core.createProjectQuestion.meArchitecture.title": "Architecture of Search Based Message Extension",
```
Comment (supkasar) [inline packages/fx-core/resource/package.nls.json:391]: Architecture of Search-Based Message Extension

Comment (yuqizhou77) [reply packages/fx-core/resource/package.nls.json:391]: just checked the doc and the term is "search-based message extension"
image

**Pull request merged**